doc/man/8/rbdmap.rst

```diff
@@ -16,16 +16,86 @@ Synopsis
 Description
 ===========
 
-**rbdmap** is a shell script that can be run manually by the system
-administrator at any time, or automatically at boot time by the init system
-(sysvinit, upstart, systemd). The script looks for an environment variable
-``RBDMAPFILE``, which defaults to ``/etc/ceph/rbdmap``. This file is
-expected to contain a list of RBD images and, possibly, parameters to be
-passed to the underlying ``rbd`` command. The syntax of
-``/etc/ceph/rbdmap`` is described in the comments at the top of that file.
-
-The script mounts devices after mapping, and unmounts them before
-unmapping.
+**rbdmap** is a shell script that automates ``rbd map`` and ``rbd unmap``
+operations on one or more RBD (RADOS Block Device) images. While the script can be
+run manually by the system administrator at any time, the principal use case is
+automatic mapping/mounting of RBD images at boot time (and unmounting/unmapping
+at shutdown), as triggered by the init system (a systemd unit file,
+``rbdmap.service`` is included with the ceph-common package for this purpose).
+
+The script takes a single argument, which can be either "map" or "unmap".
+In either case, the script parses a configuration file (defaults to ``/etc/ceph/rbdmap``,
+but can be overrided via an  environment variable ``RBDMAPFILE``). Each line
+of the configuration file corresponds to an RBD image which is to be mapped, or
+unmapped.
+
+The configuration file format is::
+
+    IMAGESPEC RBDOPTS
+
+where ``IMAGESPEC`` should be specified as ``POOLNAME/IMAGENAME`` (the pool
+name, a forward slash, and the image name), or merely ``IMAGENAME``, in which
+case the ``POOLNAME`` defaults to "rbd". ``RBDOPTS`` is an optional list of
+parameters to be passed to the underlying ``rbd map`` command. These parameters
+and their values should be specified as a comma-separated string::
+
+    PARAM1=VAL1,PARAM2=VAL2,...,PARAMN=VALN 
+
+This will cause the script to issue an ``rbd map`` command like the following::
+
+    rbd map POOLNAME/IMAGENAME --PARAM1 VAL1 --PARAM2 VAL2 
+
+(See the ``rbd`` manpage for a full list of possible options.)
+
+When run as ``rbdmap map``, the script parses the configuration file, and for
+each RBD image specified attempts to first map the image (using the ``rbd map``
+command) and, second, to mount the image.
+
+When run as ``rbd unmap``, the script parses the configuration file, and
+attempts to first unmount, and then unmap, each image specified.
+
+If successful, the ``rbd map`` operation maps the image to a ``/dev/rbdX``
+device, at which point a udev rule is triggered to create a symlink,
+``/dev/rbd/POOLNAME/IMAGENAME``, pointing to the real mapped device.
+
+In order for mounting/unmounting to succeed, the "pretty" device name must
```

Comment:
Super nit: "pretty" sounds off to me, plus it's the first mention of pretty.  Perhaps refer to it as the friendly device name and on line 58, state "... is triggered to create a friendly device name symlink, ..."